### PR TITLE
e2e: refactor importable workflow call to support chain upgrades

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

See here for e2e workflows used outside of this repo:

https://github.com/chatton/ibc-go-e2e-demo/blob/master/.github/workflows/e2e-upgrade.yml

Workflow run: https://github.com/chatton/ibc-go-e2e-demo/actions/runs/4196041079

closes: #2501 

### Commit Message / Changelog Entry

```bash
e2e: refactor importable workflow call to support chain upgrades
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testsuite/testsuite.go [hunk 1]
```diff
@@ -172,6 +172,28 @@ func (s *E2ETestSuite) SetupChainsRelayerAndChannel(ctx context.Context, channel
 	return r, chainAChannels[len(chainAChannels)-1]
 }
 
+// TODO: Actually setup a single chain.
+// Seeing panic: runtime error: index out of range [0] with length 0 when using a single chain.
+// https://github.com/strangelove-ventures/interchaintest/issues/401
+func (s *E2ETestSuite) SetupSingleChain(ctx context.Context) *cosmos.CosmosChain {
```
Comment (damiannolan) [inline e2e/testsuite/testsuite.go:178]: Unfortunately having to spin up two chains here, even tho the second is essentially discarded. I've opened an issue on interchaintest.

File: cmd/build_test_matrix/main.go [hunk 2]
```diff
@@ -102,6 +104,11 @@ func getGithubActionMatrixForTests(e2eRootDirectory, suite string, exlcudedItems
 			return fmt.Errorf("failed extracting test suite name and test cases: %s", err)
 		}
 
+		testName := os.Getenv(testNameEnv)
+		if testName != "" && contains(testName, testCases) {
```
Comment (colin-axner) [inline cmd/build_test_matrix/main.go:108]: Should we return an error if the test name provided doesn't exist?

**Review by colin-axner — APPROVED** [review]
Nice work! LGTM :magic_wand:

File: cmd/build_test_matrix/main.go (hunk 2 above)
Comment (damiannolan) [reply cmd/build_test_matrix/main.go:108]: nice catch, updated!

File: e2e/testsuite/testsuite.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (damiannolan) [reply e2e/testsuite/testsuite.go:179]: Whoops! Added :)

File: cmd/build_test_matrix/main.go (hunk 2 above)
Comment (colin-axner) [reply cmd/build_test_matrix/main.go:108]: An empty testName is valid? Should the else just be `else if testName != ""`?

Comment (damiannolan) [reply cmd/build_test_matrix/main.go:108]: whooops lol

**Pull request merged**